plugins/woocommerce/src/Internal/Admin/Logging/Settings.php

```diff
@@ -0,0 +1,450 @@
+use Automattic\WooCommerce\Internal\Traits\AccessiblePrivateMethods;
+use WC_Admin_Settings;
+use WC_Log_Handler, WC_Log_Handler_DB, WC_Log_Handler_File, WC_Log_Levels;
+
+/**
+ * Settings class.
+ */
+class Settings {
+
+	use AccessiblePrivateMethods;
+
+	/**
+	 * Default values for logging settings.
+	 *
+	 * @const array
+	 */
+	private const DEFAULTS = array(
+		'logging_enabled'           => true,
+		'default_handler'           => LogHandlerFileV2::class,
+		'retention_period_days'     => 30,
+		'level_threshold'           => 'none',
+		'file_entry_collapse_lines' => true,
+	);
+
+	/**
+	 * The prefix for settings keys used in the options table.
+	 *
+	 * @const string
+	 */
+	private const PREFIX = 'woocommerce_logs_';
+
+	/**
+	 * Class Settings.
+	 */
+	public function __construct() {
+		self::add_action( 'wc_logs_load_tab', array( $this, 'save_settings' ) );
+	}
+
+	/**
+	 * The definitions used by WC_Admin_Settings to render and save settings controls.
+	 *
+	 * @return array
+	 */
+	private function get_settings_definitions(): array {
+		$settings = array(
+			'start'                 => array(
+				'title' => __( 'Logs settings', 'woocommerce' ),
+				'id'    => self::PREFIX . 'settings',
+				'type'  => 'title',
+			),
+			'logging_enabled'       => array(
+				'title'    => __( 'Logger', 'woocommerce' ),
+				'desc'     => __( 'Enable logging', 'woocommerce' ),
+				'id'       => self::PREFIX . 'logging_enabled',
+				'type'     => 'checkbox',
+				'value'    => $this->logging_is_enabled() ? 'yes' : 'no',
+				'default'  => self::DEFAULTS['logging_enabled'] ? 'yes' : 'no',
+				'autoload' => false,
+			),
+			'default_handler'       => array(),
+			'retention_period_days' => array(),
+			'level_threshold'       => array(),
+			'end'                   => array(
+				'id'   => self::PREFIX . 'settings',
+				'type' => 'sectionend',
+			),
+		);
+
+		if ( true === $this->logging_is_enabled() ) {
+			$settings['default_handler']       = $this->get_default_handler_setting_definition();
+			$settings['retention_period_days'] = $this->get_retention_period_days_setting_definition();
+			$settings['level_threshold']       = $this->get_level_threshold_setting_definition();
+		}
+
+		$default_handler = $this->get_default_handler();
+		if ( in_array( $default_handler, array( LogHandlerFileV2::class, WC_Log_Handler_File::class ), true ) ) {
+			$settings += $this->get_filesystem_settings_definitions();
+		} elseif ( WC_Log_Handler_DB::class === $default_handler ) {
+			$settings += $this->get_database_settings_definitions();
+		}
+
+		return $settings;
+	}
+
+	/**
+	 * The definition for the default_handler setting.
+	 *
+	 * @return array
+	 */
+	private function get_default_handler_setting_definition(): array {
+		$handler_options = array(
+			LogHandlerFileV2::class  => __( 'File system (default)', 'woocommerce' ),
+			WC_Log_Handler_DB::class => __( 'Database (not recommended on live sites)', 'woocommerce' ),
```

Comment:
why is this not recommended for live sites?

Reply:
Performance reasons. Getting a lot of simultaneous database writes for logging stuff, while also handling other reads and writes could be problematic, especially for high traffic sites. There's a discussion about it here: p9o2xV-3wh-p2

Reply:
Makes sense, thanks!